### DISCUSSION TITLE
When using sofa to carve a soft body, the cut part will fall off. Is it possible to make the cut part fall off when carving a rigid body?

### DISCUSSION THREAD

**Discussion by MingSongJiang** [InfinyTech3D/SofaUnity#124] [Help - Using SofaUnity]
Hello,
When using sofa to carve a soft body, the cut part will fall off. Is it possible to make the cut part fall off when carving a rigid body?
Can sofa determine whether the two parts of the carved object are completely separated?
How to judge whether the object has been completely cut?
Thanks!

**Comment by epernod** [comment]
Hi,

no, for the moment no component has been developed to compute that.
I'm not sure to understand, do you want the cut part to fall off? do you want to delete it? or remove it with device as part of the operation?

**Reply by MingSongJiang** [reply]
Hi,
I want the completely separated parts to fall freely to the ground.
The current situation is that when the object being cut is a soft body, the completely separated part will fall, but if it is a rigid body, the completely separated part will remain in the original place and will not fall to the ground.
Thanks!

**Reply by epernod** [reply]
Hi,

normally if you cut it, it should fall. Maybe some part of your mesh are fixed in the air. Is there any FixedConstraint in your model?

Best regards
